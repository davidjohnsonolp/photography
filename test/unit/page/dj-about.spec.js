import '../../../public/src/page/dj-about.js';

let aboutPage;

describe('DJAbout About Page', () => {
  beforeEach(() => {
    aboutPage = window.document.createElement('dj-about'); 
    document.body.appendChild(aboutPage);
  });

  afterEach(() => {
    document.body.getElementsByTagName('dj-about')[0].remove();
  });

  it('renders correctly', () => {
    expect({ html: aboutPage.shadowRoot.innerHTML }).toMatchSnapshot();
  });
});