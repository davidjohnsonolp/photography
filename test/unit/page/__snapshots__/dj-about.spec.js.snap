// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DJAbout About Page renders correctly 1`] = `
Object {
  "html": "
      <style>
        :host {
          display: flex;
          flex-direction: column;
          flex: 1 0 auto;
        }
        main {
          background: white;
          overflow: hidden;
          flex: 1 0 auto;
          padding: 0 5px;
        }
        @media screen and (min-width: 735px) {
          dj-figure {
            float: right;
            width: 33%;
          }
        }
      </style>
      <dj-header>
      </dj-header>

      <main>
        <dj-h2>About David Johnson</dj-h2>
        <dj-figure src=\\"https://pbs.twimg.com/media/CqAdEzeXYAAlpub.jpg:small\\" alt=\\"First shoot on the River Clyde\\">
          First shoot on the River Clyde
        </dj-figure>
        <dj-p>
            This is an online resource for amateur photgrapher David Johnson. The goal of the website is showcase various pictures of landscapes in Scotland and boyond. 
        </dj-p>
        <dj-p>
            Starting in August 2016 and based in the city of Glasgow, David aims to capture all the beauty Scotland has to offer - whether it's in the inner city or on top of a Munro.
        </dj-p>
        <dj-p>
            As the journey continues, David aims to share more photos in the hope to build up a collection documenting the one of the
            most beautiful countries on Earth. 
        </dj-p>
      </main>

      <dj-footer>
      </dj-footer>
    ",
}
`;
